src/mono/wasm/runtime/driver.c

```diff
@@ -645,6 +645,17 @@ mono_wasm_string_from_js (const char *str)
 		return NULL;
 }
 
+EMSCRIPTEN_KEEPALIVE MonoString *
+mono_wasm_string_from_utf16 (const mono_unichar2 * chars, int length)
+{
+	assert (length >= 0);
```

Comment:
Why are we asserting the length is nonzero but accepting a null pointer for `chars`?

Reply:
The existing API accepted a null pointer, so I'm carrying that forward. I'm just rejecting a negative length since that's always nonsense. I'd make the arg uint32_t but JS doesn't have uints.

Reply:
Oh I misread, apologies. Thought it said `length > 0`.